src/NodaTime/Calendars/Wondrous/WondrousCalendarHelper.cs

```diff
@@ -0,0 +1,2204 @@
+        /// <returns></returns>
+        public static LocalDateTime CreateDateTime(int wondrousYear, int wondrousMonth, int wondrousDay, int hour, int minute = 0, int second = 0, int millisecond = 0)
+        {
+            if (wondrousMonth == AyyamiHaMonth0)
+            {
+                // move Ayyam-i-Ha days to fall after the last day of month 18.
+                wondrousMonth = Month18;
+                wondrousDay += DaysInMonth;
+            }
+            return new LocalDateTime(wondrousYear, wondrousMonth, wondrousDay, hour, minute, second, millisecond, CalendarSystem.Wondrous);
+        }
+
+        /// <summary>
+        /// Return the month of this date. If in Ayyam-i-Ha, returns 0.
+        /// </summary>
+        /// <remarks>Deals with Ayyam-i-Ha.</remarks>
+        /// <param name="input"></param>
+        /// <returns></returns>
+        public static int WondrousMonthNum(this LocalDate input)
+        {
+            Preconditions.CheckArgument(input.Calendar == CalendarSystem.Wondrous, nameof(input), "Only valid when using the Wondrous calendar");
+
+            if (input.Month == Month18 && input.Day > DaysInMonth)
+            {
+                return 0;
+            }
+            return input.Month;
+        }
+
+        /// <summary>
+        /// Return the day of this month. 
+        /// </summary>
+        /// <remarks>Deals with days in Ayyam-i-Ha.</remarks>
+        /// <param name="input"></param>
+        /// <returns></returns>
+        public static int WondrousDayNum(this LocalDate input)
+        {
+            Preconditions.CheckArgument(input.Calendar == CalendarSystem.Wondrous, nameof(input), "Only valid when using the Wondrous calendar");
+
+            if (input.Month == Month18 && input.Day > DaysInMonth)
+            {
+                return input.Day - DaysInMonth;
+            }
+            return input.Day;
+        }
+
+
+        /// <summary>
+        /// Return the month of this date. If in Ayyam-i-Ha, returns 0.
+        /// </summary>
+        /// <remarks>Deals with Ayyam-i-Ha.</remarks>
+        /// <param name="input"></param>
+        /// <returns></returns>
+        public static int WondrousMonthNum(this LocalDateTime input)
+        {
+            return input.Date.WondrousMonthNum();
+        }
+
+        /// <summary>
+        /// Return the day of this month. 
+        /// </summary>
+        /// <remarks>Deals with days in Ayyam-i-Ha.</remarks>
+        /// <param name="input"></param>
+        /// <returns></returns>
+        public static int WondrousDayNum(this LocalDateTime input)
+        {
+            return input.Date.WondrousDayNum();
+        }
+
+        //public static WondrousYearMonthDay AsWondrousYearMonthDay(this LocalDate input)
+        //{
+        //    return new WondrousYearMonthDay(
+        //        input.Year,
+        //        input.WondrousMonthNum(),
+        //        input.WondrousDayNum());
+        //}
+
+        /// <summary>
```

Comment:
I'd remove this until we have an implementation.

Reply:
Unfortunately, something like this is needed. Because the internal representation of the date is not always what the "real" date is, the standard "ToString" can't be used.  

I didn't want to go much further down this path until I knew that the implementation would be viable.  I suspect that this will be the function most used by developers. There is not much need to "add 10 days" to the current month, etc. Most current uses will be "show me [Gregorian date] in the Wondrous calendar using this format".

Reply:
In that case, it might be worth not having the extension methods. We could have the struct with either methods for converting values, or maybe have implicit conversions.

Another thing to *consider* is putting those types in a separate NuGet package. The calendar system would be in Noda Time itself (and would have to be) but the "slightly unusual" type could be in its own package which would avoid anyone accidentally using bits without wanting to.

Maybe we should get the calculator/calendar system in first, and hold off on everything else for now? (Once anything public is in, it can't be removed - that makes me nervous about any additions, basically. Caution is my watchword.)

Reply:
Yes, that would make sense. Don't need to put a lot of code into the core system that could be in another package.